presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2169,8 +2169,7 @@ public ActionType getType()
 
         public T getData()
         {
-            checkState(type != ActionType.DROP);
-            return data;
+            return requireNonNull(data, "data is null");
```

Comment:
This pattern seems to go against our guidelines.  How about making `data` an `Optional<T>`?

Reply:
Refactored

Reply:
@aalbu see my comment https://github.com/prestosql/presto/pull/2692#discussion_r374928793:

> * We had validation that `data != null` (unless it's DROP action). We no longer have it.
> * I'm not a fan of nullable fields, but it appears that change `T data` to `Optional<T> data` is causing a lot of changes here, making the actual behavioral change invisible. I would back this out from this PR.

if we want to keep it, this needs to be a separate commit (this isn't really just few lines of cleanup), and we should retain existing validation.

Reply:
My intent wasn't to suggest removing validation, just setting the value to `Optional.empty()` instead of null. For many callers, returning an `Optional` seemed a good fit, but I didn't realize the footprint of the change.  Sorry.